release/nightly_tests/dataset/read_tfrecords_benchmark.py

```diff
@@ -0,0 +1,108 @@
+import random
+import shutil
+import tempfile
+from typing import List, Tuple
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+from read_images_benchmark import generate_images
+import pyarrow as pa
+import numpy as np
+
+
+def read_tfrecords(path: str) -> Dataset:
+    return ray.data.read_tfrecords(paths=path)
+
+
+def generate_tfrecords_from_images(
+    num_images: int, sizes: List[Tuple[int, int]], modes: List[str], formats: List[str]
+) -> str:
+    images_dir = generate_images(num_images, sizes, modes, formats)
+    try:
+        ds = ray.data.read_images(images_dir)
+
+        # Convert images from NumPy to bytes
+        def images_to_bytes(batch):
+            images_as_bytes = [image.tobytes() for image in batch]
+            return pa.table({"image": images_as_bytes})
+
+        ds = ds.map_batches(images_to_bytes, batch_format="numpy")
+        tfrecords_dir = tempfile.mkdtemp()
+        ds.write_tfrecords(tfrecords_dir)
```

Comment:
Do you plan to benchmark write_tfrecords? In this case, it looks the write bench that makes it clear the read is so much worse.

Reply:
@jianoaix - right, `write_tfrecords` can be added later, which should be simple to add given the added methods here.